src/main/java/bio/terra/cloudfunctions/common/ContentHandler.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.cloudfunctions.common;
+
+import com.google.events.cloud.storage.v1.StorageObjectData;
+
+/**
+ * This class is an abstract representation of application-specific content handler for extracting
+ * relevant information and formatting the StorageObjectData payload for inserting into BigQuery.
+ *
+ * <p>Sub-classes of this class can be integrated with a DI framework to deploy Cloud Functions like
+ * a Service.
+ */
+public abstract class ContentHandler extends FileTypeDetector {
```

Comment:
why does the handler class extend the filetypedetector? What does it gain from extending?

Reply:
The thinking here is to separate the low-level file-based IO operations (open input stream, uncompress, untar etc) that all handlers need from the `translator` and `insert` logic. The `FileTypeDetector` should also include the file path and bucket, the content-type, the triggering event type you mentioned in the previous comment.